unittest/server/src/ivi_input_controller_uinttests.cpp

```diff
@@ -0,0 +1,1714 @@
+    mpp_ctxSeat[0]->forced_surf_enabled = ILM_FALSE;
+    struct ivisurface l_ivi_surf;
+    l_ivi_surf.layout_surface = (struct ivi_layout_surface *)0xFFFFFFFF;
+    mpp_ctxSeat[0]->forced_ptr_focus_surf = &l_ivi_surf;
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus = (struct weston_view *)malloc(sizeof(struct weston_view));
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus->surface = (struct weston_surface *)0xFFFFFFFF;
+
+    struct weston_surface *l_surf[1] = {(struct weston_surface *)0xFFFFFFFF};
+    SET_RETURN_SEQ(surface_get_weston_surface, l_surf, 1);
+
+    pointer_grab_focus(&mpp_ctxSeat[0]->pointer_grab);
+
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 1);
+    ASSERT_EQ(weston_pointer_clear_focus_fake.call_count, 1);
+
+    free(mpp_ctxSeat[0]->pointer_grab.pointer->focus);
+    free(mpp_ctxSeat[0]->pointer_grab.pointer);
+}
+
+/** ================================================================================================
+ * @test_id             pointer_grab_focus_nullSurfCtx
+ * @brief               Test case of pointer_grab_focus() where input ctxSeat button_count is disable {0}
+ *                      and forced_surf_enabled is enable {1} and forced_ptr_focus_surf, pointer focus is an object
+ * @test_procedure Steps:
+ *                      -# Set input ctxSeat
+ *                      -# Mocking the surface_get_weston_surface() does return an object
+ *                      -# Calling the pointer_grab_focus()
+ *                      -# Verification point:
+ *                         +# surface_get_weston_surface() must be called once time
+ *                         +# weston_pointer_set_focus() must be called once time
+ *                         +# weston_pointer_clear_focus() not be called
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(InputControllerTest, pointer_grab_focus_nullSurfCtx)
+{
+    mpp_ctxSeat[0]->pointer_grab.pointer = (struct weston_pointer *)malloc(sizeof(struct weston_pointer));
+    mpp_ctxSeat[0]->pointer_grab.pointer->button_count = 0;
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus = (struct weston_view *)0xFFFFFFFF;
+    mpp_ctxSeat[0]->forced_surf_enabled = ILM_TRUE;
+    struct ivisurface l_ivi_surf;
+    l_ivi_surf.layout_surface = (struct ivi_layout_surface *)0xFFFFFFFF;
+    mpp_ctxSeat[0]->forced_ptr_focus_surf = &l_ivi_surf;
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus = (struct weston_view *)malloc(sizeof(struct weston_view));
+    mpp_ctxSeat[0]->pointer_grab.pointer->focus->surface = (struct weston_surface *)0xFFFFFFFF;
+
+    struct weston_surface *l_surf[1] = {(struct weston_surface *)malloc(sizeof(struct weston_surface))};
+    custom_wl_list_init(&l_surf[0]->views);
+    SET_RETURN_SEQ(surface_get_weston_surface, l_surf, 1);
+
+    pointer_grab_focus(&mpp_ctxSeat[0]->pointer_grab);
+
+    ASSERT_EQ(surface_get_weston_surface_fake.call_count, 1);
+    ASSERT_EQ(weston_pointer_set_focus_fake.call_count, 1);
```

Comment:
In this case, the second argument of weston_pointer_set_focus should not be null